src/zkey/cca.c

```diff
@@ -97,6 +110,17 @@ static int get_cca_version(struct cca_lib *cca, bool verbose)
 
 	util_assert(cca != NULL, "Internal error: cca is NULL");
 
+	if (cca->lib_csulcca == NULL) {
+		pr_verbose(verbose, "CCA host library not available.");
+		return -ELIBACC;
+	}
+
+	if (cca->dll_CSUACFV == NULL) {
```

Comment:
Can this really happen? Aren't you checking for all the functions being not NULL in load_cca_library() ? 
So when lib_csulcca is not NULL, then the functions should also all be there, no?

Reply:
The idea here was to in fact allow single functions to be NULL and always check the function ptr "at use". I agree that in the past it was like you say "lib_csulcca is not NULL, then the functions should also all be there". But this may change as we introduce more functions from newer libs, but customers still use older ones.